.circleci/config.yml

```diff
@@ -154,18 +162,36 @@ commands:
           paths:
             - ".local-*"
 
-  prepare-python:
+  prepare-pip-cache:
     parameters:
       py-version:
         <<: *py-version-template
     steps:
-      - attach-and-link:
-          py-version: << parameters.py-version >>
-      - conditional-skip
-      - config-path
       - restore_cache:
+          keys:
+          - pip-cache-{{ arch }}-<< parameters.py-version >>
+          - pip-cache-{{ arch }}-
+          - pip-cache-
+      - run:
+          name: Downloading dependencies
+          command: pip download --dest /tmp -c constraints-dev.txt -r requirements-circleci.txt
```

Comment:
This will download the packages to `--dest` and warm the cache at the same time, I'm not interested in the actual `.whl`, just on the cache warming as a side-effect.

Reply:
Hm I don't get what the advantage of that is.

AFAICS this only lengthens runtime because now pip will check every dependency twice per container run.

Reply:
Because:

> I want to add a job in the build to make sure our constraints.txt does not contain unnecessary dependencies, this is just the ground work and should not change existing behavior.

So I just want a command to warm the cache, and then I want to add a job to run `pip install -r requirements.txt -c constraints.txt` and compare the `pip freeze` with the existing `constraints.txt`. If the `constraints.txt` has more entries it means that we or a downstream dependency removed a dependency, and the `constraints.txt` should be cleared.

Note: I'm thinking about having separate `constraints.txt` for different OSs, otherwise this does not work.

Reply:
I did not want to use `pip install` because that would pollute the `pip freeze`. However, I can rollback this change to use `pip install` and just generate a new venv if you want.